src/ffi/extra/testing.cc

```diff
@@ -159,7 +160,7 @@ class TestUnregisteredObject : public TestUnregisteredBaseObject {
                               TestUnregisteredBaseObject);
 };
 
-TVM_FFI_NO_INLINE void TestRaiseError(String kind, String msg) {
+TVM_FFI_NO_INLINE void TestRaiseError(const String& kind, const String& msg) {
```

Comment:
for the ffi registered functions, given we already ask for a fresh value that can be moved, pass by value is better, and allows small string be passed by stack